android/src/main/java/com/swmansion/reanimated/nodes/Node.java

```diff
@@ -93,7 +95,7 @@ protected final void forceUpdateMemoizedValue(T value) {
     markUpdated();
   }
 
-  private static void findAndUpdateNodes(Node node, Set<Node> visitedNodes) {
+  private static void findAndUpdateNodes(Node node, Set<Node> visitedNodes, Deque<FinalNode> finalsToBeUpdated) {
```

Comment:
rename -> finalNodesQueue

Also I think you want to just use Stack instead of Deque